### PR TITLE
Issue #8364: Add support for enhanced instanceof syntax to HiddenFieldCheck(#7290)

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8364: Add support for enhanced instanceof syntax to HiddenFieldCheck(#7290)

~~Reviewers, please note that the first commit is from the instanceof PR, #8401.~~
Commit has been dropped now that #8401 was merged.

Check regression report: https://checkstyle-reports-java14.s3.amazonaws.com/reports/diff_instanceof-HiddenFieldCheck/index.html

### CODE REVIEW

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.checks.coding.hiddenfield;
+
+import java.util.Locale;
+
+public class InputHiddenFieldEnhancedInstanceof {
```
Comment (romani) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java:6]: please guess what is missed over class and in code 

Can we make instance of multliline regexp Check to catch that all Inputs has `// config:` and `// ok` or `// violation` ... all existing violations in code .... add to suppression with comment `utill ....` ?

Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java:6]: All comments in input files should only be one of three cases:

1. ` // config:`
2. `// ok`
3. `// violation`

Is that correct?

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java:6]: yes, and please exclude all Inputs from `grammar` folder, they will not have such comments.

Comment (romani) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java:7]: multiline comment please

Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/hiddenfield/InputHiddenFieldEnhancedInstanceof.java:14]: Fixed.

**Pull request merged**